.github/actions/validate-deployment-settings/action.yml

```diff
@@ -0,0 +1,106 @@
+name: Validate Environment Deployment Settings
+description: Action that validates various `ACCESS-NRI/build-cd` deployment settings
+inputs:
+  settings-path:
+    type: string
+    required: false
+    default: './cd/config/settings.json'
+    description: Path to the build-cd settings.json file to validate
+  target:
+    type: string
+    required: false
+    default: Gadi
+    description: Check settings for this Deployment Target
+  error-level:
+    type: string
+    required: false
+    default: warning
+    description: Whether failed validation checks should be 'notice', 'warning' or 'error'
+outputs:
+  failures:
+    value: ${{ steps.collector.outputs.failures }}
+    description: Comma-delimited string of failures encountered during validation
+runs:
+  using: composite
+  steps:
+    - name: Check All Spack Major Versions in Prerelease are in Release
+      id: spack-maj-all-pre-in-rel
+      shell: bash
+      run: |
+        majors_only_in_prerelease=$(jq --compact-output --raw-output \
+          '.deployment.${{ inputs.target }} | (.Prerelease.spack | keys) - (.Release.spack | keys) | @sh' \
+          ${{ inputs.settings-path }}
+        )
+
+        if [[ "$majors_only_in_prerelease" != "" ]]; then
+          msg="Spack versions ${majors_only_in_prerelease} do not exist in Release. Deployments to Release at those versions will not succeed."
+          echo "::${{ inputs.error-level }}::$msg"
+          echo "msg=$msg" >> $GITHUB_OUTPUT
+        fi
```

Comment:
It's warning the user that there is a Prerelease at version 0.XX that is not in Release. Which is fine (we could be just testing a new version of spack and may not use it) but it will error out if you go to merge the PR.